### ISSUE TITLE
Enabling of pydantic support makes typing.Protocol requires imports in Runtime.

### ISSUE THREAD

**Issue by OlehChyhyryn** [snok/flake8-type-checking#180]
Take the following code snippet.

If Pydantic support is disabled, it requires only. 

```
from typing import TYPE_CHECKING, Protocol

if TYPE_CHECKING:
    from django.db.models.query import QuerySet


class ModelProtocol(Protocol):
    objects: QuerySet
```

With Pydantic support disabled, there are no mistakes, but after enabling it errors occur:

![image](https://github.com/snok/flake8-type-checking/assets/39335857/670c9869-6ffb-46c2-8467-001c0f22b6a6)

Some information is hidden from privacy reasoning.

**Comment by Daverball** [comment]
That's expected behavior with pydantic support enabled. Since a pydantic model requires all the annotations to be available at runtime. Fully tracking what is and isn't a pydantic model or used inside a pydantic model is pretty much impossible without a type checker plugin, so this is the trade-off you get with that mode.

That's what the `type-checking-pydantic-enabled-baseclass-passlist` setting is for, you can mark any base classes that for sure aren't pydantic models, so you could add `Protocol` to that pass list.

**Comment by Daverball** [comment]
That being said we could change the pydantic mode to act more like the SQLAlchemy mode, in that it records these as soft-uses, rather than hard-uses. So it would generate neither a false positive for TC004 nor TC001-003 in the opposite case.
